external/anomaly/tests/test_ote_training.py

```diff
@@ -0,0 +1,339 @@
+                                                       OTETestPotEvaluationAction,
+                                                       OTETestNNCFAction,
+                                                       OTETestNNCFEvaluationAction,
+                                                       OTETestNNCFExportAction,
+                                                       OTETestNNCFExportEvaluationAction)
+
+
+logger = get_logger(__name__)
+
+def DATASET_PARAMETERS_FIELDS() -> List[str]:
+    return deepcopy(['dataset_path',
+                     ])
+
+DatasetParameters = namedtuple('DatasetParameters', DATASET_PARAMETERS_FIELDS())
+
+
+def _get_dataset_params_from_dataset_definitions(dataset_definitions, dataset_name):
+    if dataset_name not in dataset_definitions:
+        raise ValueError(f'dataset {dataset_name} is absent in dataset_definitions, '
+                         f'dataset_definitions.keys={list(dataset_definitions.keys())}')
+    cur_dataset_definition = dataset_definitions[dataset_name]
+    training_parameters_fields = {k: v for k, v in cur_dataset_definition.items()
+                                  if k in DATASET_PARAMETERS_FIELDS()}
+    print(f"training_parameters_fields: {training_parameters_fields}")
+    make_paths_be_abs(training_parameters_fields, dataset_definitions[ROOT_PATH_KEY])
+    print(f"training_parameters_fields after make_paths_be_abs: {training_parameters_fields}")
+
+    assert set(DATASET_PARAMETERS_FIELDS()) == set(training_parameters_fields.keys()), \
+            f'ERROR: dataset definitions for name={dataset_name} does not contain all required fields'
+    assert all(training_parameters_fields.values()), \
+            f'ERROR: dataset definitions for name={dataset_name} contains empty values for some required fields'
+
+    params = DatasetParameters(**training_parameters_fields)
+    return params
+
+
+def _create_anomaly_classification_dataset_and_labels_schema(dataset_params):
+    logger.debug(f'Path to dataset: {dataset_params.dataset_path}')
+    category_list = [f.path for f in os.scandir(dataset_params.dataset_path) if f.is_dir()]
+    items = []
+    for category in category_list:
+        if "vivo" not in category:
+            logger.debug(f'Creating dataset for {category}')
+            items.extend(OteMvtecDataset(path=category, seed=0).generate())
+    dataset = DatasetEntity(items=items)
+    labels = dataset.get_labels()
+    labels_schema = LabelSchemaEntity.from_labels(labels)
+    return dataset, labels_schema
+
+
+def get_anomaly_test_action_classes() -> List[Type[BaseOTETestAction]]:
+    return [
+        AnomalyDetectionTestTrainingAction,
+        OTETestTrainingEvaluationAction,
+        OTETestExportAction,
+        OTETestExportEvaluationAction,
+        OTETestPotAction,
+        OTETestPotEvaluationAction,
+        OTETestNNCFAction,
+        OTETestNNCFEvaluationAction,
+        OTETestNNCFExportAction,
+        OTETestNNCFExportEvaluationAction,
```

Comment:
Please, add `OTETestNNCFGraphAction` -- it was added recently by Alexander Dokuchaev.

Reply:
Will add after nncf implementation.

Reply:
Please, add `OTETestNNCFGraphAction` and make it xfail

Reply:
Added. Please see latest job 1323.